### PR TITLE
docs: Improve helm docs #15706

### LINKED ISSUE #15706: Helm + values + valueFiles not working as expected
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
Hey, we currently use kustomize and are giving the helm approach a go

application manifest
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: ${name}-${environment}
  namespace: argocd
spec:
  project: ${environment}
  source:
    repoURL: https://github.com/xxx
    targetRevision: feature/helm-app
    path: helm/app
    helm:
      values: |-
        nameOverride:   I_AM_MIDDLE_OVERRIDE
      valueFiles:
        # TODO: this doesn't take precedence over ^
        - ../../manifests/${environment}/values.yaml
```    
../../manifests/${environment}/values.yaml
```
nameOverride: I_AM_FINAL_OVERRIDE
```

helm/app/values.yaml
```
nameOverride: I_AM_DEFAULT
```

i expected the following to result in a result similar to running 
```
kubectl get Applications  ${name}-${environment}-o jsonpath="{.spec.source.helm.values}" -n argocd > ./argocd-values.yml
helm template --debug ./helm/app -f  ./argocd-values.yml -f ../../manifests/${environment}/values.yaml
```

as in the merge order  i want to be used is
* /helm/app/values.yaml 
* ./argocd-values.yml
* ../../manifests/${environment}/values.yaml

But what i see is `./argocd-values.yml ` takes precedence over `../../manifests/${environment}/values.yaml`
or rather `helm.values` takes precedence over `valueFiles`
or rather `nameOverride:  I_AM_MIDDLE_OVERRIDE` is used for resources

But then in the ui the parameters section instead shows  `I_AM_FINAL_OVERRIDE`, But it isnt used.

 if i set an override value to anything it is then used and reflected on resources

What is the correct way this should be working?
should the parameter section reflect what should be getting used? 

or am i miss understanding something

**Version**

```shell
v2.8.4
```

Thanks!

### PR DESCRIPTION
**Pull request by phyzical**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Just adds some simple notes around helm and how the order of precedence occurs, as in which values win.

Also added a note around the fact that the ui isn't fully obvious whats used with a caveat to the issue that should solve this ambiguity at some point, and a workaround to avoid this

closes #15706

### CODE REVIEW

File: docs/user-guide/helm.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ spec:
     namespace: kubeseal
 ```
 
+!!! note "When using multiple methods"
+    Order of precedence is `parameters > valuesObject > values > valueFiles > helm repository values.yaml` (see [this stackoverflow](https://stackoverflow.com/questions/76739856/what-is-the-precedence-setting-of-various-helm-values-in-an-argocd-application-m) for a more detailed example)
```
Comment (ishitasequeira) [inline docs/user-guide/helm.md:29]: Could we add the example here directly, instead of linking to stackoverflow?
cc @crenshaw-dev

**Review by ishitasequeira — DISMISSED** [review]

File: docs/user-guide/helm.md (hunk 1 above)
Comment (phyzical) [reply docs/user-guide/helm.md:29]: happy to do that,

Just felt it was a bit loud how about we reduce what the user sees for 
```
(see [this stackoverflow](https://stackoverflow.com/questions/76739856/what-is-the-precedence-setting-of-various-helm-values-in-an-argocd-application-m) for a more detailed example)
```
to
```
[Detailed Example](https://stackoverflow.com/questions/76739856/what-is-the-precedence-setting-of-various-helm-values-in-an-argocd-application-m)
```

Comment (blakepettersson) [reply docs/user-guide/helm.md:29]: The issue with having a link is that the information can get stale, or the site might be down, or the link might not exist anymore. Having an example directly in the docs is also more easy to parse for a user than needing to dig through a thread (even if in this particular case it's just one answer).

Comment (todaywasawesome) [reply docs/user-guide/helm.md:29]: Agreed @phyzical, lets keep docs more self-contained.

Comment (phyzical) [reply docs/user-guide/helm.md:29]: no worries will update now 👍

**Review by blakepettersson — APPROVED** [review]
LGTM 👍

**Comment by blakepettersson** [comment]
@phyzical can you signoff your latest commit?

```
git commit --amend -s
```

**Comment by blakepettersson** [comment]
Now it seems like a weird commit got into this PR (0ddae4b)? Could you remove that one from the PR @phyzical?

**Pull request merged**